cmd/cloud-controller-manager/app/controllermanager.go

```diff
@@ -208,12 +208,22 @@ func startControllers(c *cloudcontrollerconfig.CompletedConfig, stop <-chan stru
 	nodeController := cloudcontrollers.NewCloudNodeController(
 		c.SharedInformers.Core().V1().Nodes(),
 		client("cloud-node-controller"), cloud,
-		c.ComponentConfig.KubeCloudShared.NodeMonitorPeriod.Duration,
 		c.ComponentConfig.NodeStatusUpdateFrequency.Duration)
 
 	nodeController.Run(stop)
 	time.Sleep(wait.Jitter(c.ComponentConfig.Generic.ControllerStartInterval.Duration, ControllerStartJitter))
 
+	cloudNodeGarbageCollectorController, err := cloudcontrollers.NewCloudNodeGarbageCollectorController(
```

Comment:
since you mentioned naming I did some brainstorming :)
- cloudHasNodeController
- cloudNodeOnController
- cloudNodeThereController
- cloudNodePresentController
- cloudNodePresenceController
- cloudNodeRosterController
- cloudNodeFinder
- cloudNodeChecker

When I was hacking around with this, I got in the Halloween spirit and called it the NodeReaper... but that's really no better than GarbageCollector...

But GarbageCollector doesn't really fit because this controller can also just taint.

Maybe something like `cloudNodeExistsController` would work? Because the point is to check whether the node is still there, even though the Kubelet isn't reporting? (and the three modes are "there and not working," "there and shut down," or "not there")

Reply:
What are your thoughts on having both a `CloudNodeLifecycleController` and a `NodeLifeCycleController`. I like the term "LifeCycle" as I think its accurate as to what the controller is doing but maybe it might be confusing to have controllers with similar names.

Reply:
sgtm